sfge/src/main/java/com/salesforce/rules/unusedmethod/RuleStateTracker.java

```diff
@@ -0,0 +1,128 @@
+package com.salesforce.rules.unusedmethod;
+
+import com.salesforce.apex.jorje.ASTConstants.NodeType;
+import com.salesforce.collections.CollectionUtil;
+import com.salesforce.graph.Schema;
+import com.salesforce.graph.build.CaseSafePropertyUtil.H;
+import com.salesforce.graph.vertex.InvocableWithParametersVertex;
+import com.salesforce.graph.vertex.MethodVertex;
+import com.salesforce.graph.vertex.SFVertexFactory;
+import com.salesforce.graph.vertex.UserClassVertex;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+
```

Comment:
Individual fields'/methods' doc looks great. Could you also add javadoc for the class so that one day we can rely on generating javadoc for the project to understand how it works?